core/core.go

```diff
@@ -250,6 +257,32 @@ func initPrivateKey(path string) (p2pcrypto.PrivKey, error) {
 	return nil, err
 }
 
+func initNetworkId(networkID int, meshDB *meshdb.MeshDB) error {
+	var metadata meshdb.Metadata
+	err := meshDB.Metadata.FindByID([]byte{0}, &metadata)
```

Comment:
After looking at this again, I think it is a leaky abstraction to expose the fact that we are using `[]byte{0}` for the ID in the `core` package. We had similar concerns about the use of the special separator `"|"` in the `Orders` collection, and to address those concerns we added some helper methods to the `meshdb` package (e.g. `FindOrdersByMakerAddressAndMaxSalt` and `FindOrdersLastUpdatedBefore`). These methods hide the details about the structure of keys in the database and make the `meshdb` package easier to use.

Can we take a similar approach here and create two helper function in the `meshdb` pacakge?

- `GetMetadata` returns the metadata (or a `db.NotFoundError` if no metadata has been found).
- `SaveMetadata` inserts the metadata into the database.

Reply:
Agreed -- is a bit smelly to see in `core`.

Reply:
I'm going to push a commit with the `SaveMetadata` with the following signature, LMK if you would rather accept `metadata` as the argument. I figure either implementation is relatively easy to extend with future metadata. 

```go
func (m *MeshDB) SaveMetadata(networkID int) error
```